File: 3.Hotel Booking.cs
using System;

namespace HotelBooking
{
    class HotelBooking
    {
        static void Main(string[] args)
        {
            try
            {
                Console.Write("Enter month (words): ");
                string month = Console.ReadLine().ToLower();
                while (!("may".Equals(month)) && !("june".Equals(month)) && !("july".Equals(month)) &&
                    !("august".Equals(month)) && !("september".Equals(month)) && !("october".Equals(month)) &&
                    !("december".Equals(month)))
                {
                    Console.Write("Invalid month or bad spelling! Try again! " +
                            "Keep in mind that we offer rooms only for " +
                            "May, June, July, August, September, October and December" +
                            "(the first letter could be both capital or low): ");
                    month = Console.ReadLine().ToLower();
                }

                Console.Write("Enter number of nights (1-200): ");
                int nights = int.Parse(Console.ReadLine());
                while (nights < 1 || nights > 200)
                {
                    Console.Write("Invalid number of nights! It must be between 1 and 200. Try Again: ");
                    nights = int.Parse(Console.ReadLine());
                }


                double priceStudio = 0;
                    double priceDouble = 0;
                    double priceSuite = 0;

                    //price per a room in different months
                    switch (month)
                    {
                        case "may":
                        case "october":
                            priceStudio = 50;
                            priceDouble = 65;
                            priceSuite = 75;
                            break;
                        case "june":
                        case "september":
                            priceStudio = 60;
                            priceDouble = 72;
                            priceSuite = 82;
                            break;
                        case "july":
                        case "august":
                        case "december":
                            priceStudio = 68;
                            priceDouble = 77;
                            priceSuite = 89;
                            break;
                    }

                    //discounts per a room according to months and number of nights
                    if ((month == "may" || month == "october") && nights > 7)
                    {
                        priceStudio -= 0.05 * priceStudio;
                    }

                    if ((month == "june" || month == "september") && nights > 14)
                    {
                        priceDouble -= 0.10 * priceDouble;
                    }

                    if ((month == "july" || month == "august" || month == "december") && nights > 14)
                    {
                        priceSuite -= 0.15 * priceSuite;
                    }

                    //total prices with the discounts
                    double totalPriceStudio = priceStudio * nights;
                    double totalPriceDouble = priceDouble * nights;
                    double totalPriceSuite = priceSuite * nights;

                    //free night discount
                    if ((month == "september" || month == "october") && nights > 7)
                    {
                        totalPriceStudio = priceStudio * (nights - 1);
                    }

                    Console.WriteLine($"Studio: {totalPriceStudio:F2} lv.");
                    Console.WriteLine($"Double: {totalPriceDouble:F2} lv.");
                    Console.WriteLine($"Suite: {totalPriceSuite:F2} lv.");

            }
            catch
            {
                Console.WriteLine("Sorry, invalid input!!!Start from the beginning, please!" +
                    "Read the instructions carefully!");
            }
        }
    }
}

File: 3.Hotel Booking.java
import java.util.Scanner;

public class HotelBooking {
    public static void main(String[] args) {

        Scanner scanner = new Scanner(System.in);

        try {
            System.out.print("Enter month (words): ");
            String month = scanner.nextLine().toLowerCase();
            while (!("may".equals(month)) && !("june".equals(month)) && !("july".equals(month)) &&
                    !("august".equals(month)) && !("september".equals(month)) && !("october".equals(month)) &&
                    !("december".equals(month))) {
                System.out.print("Invalid month or bad spelling! Try again! " +
                        "Keep in mind that we offer rooms only for " +
                        "May, June, July, August, September, October and December" +
                        "(the first letter could be both capital or low): ");
                month = scanner.nextLine().toLowerCase();
            }

            System.out.print("Enter number of nights (1-200): ");
            int nightsCount = Integer.parseInt(scanner.nextLine());
            while (nightsCount < 1 || nightsCount > 200) {
                System.out.print("Invalid number of nights! It must be between 1 and 200. Try Again: ");
                nightsCount = Integer.parseInt(scanner.nextLine());
            }

            double studioPrice = 0;
            double doublePrice = 0;
            double suitePrice = 0;

            //price per a room in the different months
            if ("may".equals(month) || "october".equals(month)) {
                studioPrice = 50;
                doublePrice = 65;
                suitePrice = 75;

            } else if ("june".equals(month) || "september".equals(month)) {
                studioPrice = 60;
                doublePrice = 72;
                suitePrice = 82;

            } else if ("july".equals(month) || "august".equals(month) || "december".equals(month)) {
                studioPrice = 68;
                doublePrice = 77;
                suitePrice = 89;
            }

            //discount per a room according to a month
            if (nightsCount > 7 && ("may".equals(month) || "october".equals(month))) {
                studioPrice -= 0.05 * studioPrice;
            }

            if (nightsCount > 14 && ("june".equals(month) || "september".equals(month))) {
                doublePrice -= 0.10 * doublePrice;
            }

            if (nightsCount > 14 && ("july".equals(month) || "august".equals(month) || "december".equals(month))) {
                suitePrice -= 0.15 * suitePrice;
            }

            //total prices with the discounts
            double priceStudioWithDiscount = studioPrice * nightsCount;
            double priceDoubleWithDiscount = doublePrice * nightsCount;
            double priceSuiteWithDiscount = suitePrice * nightsCount;

            //free night discount
            if (nightsCount > 7 && ("September".equals(month) || "October".equals(month))) {
                priceStudioWithDiscount = studioPrice * (nightsCount - 1);
            }

            System.out.printf("Studio: %.2f lv.\n", priceStudioWithDiscount);
            System.out.printf("Double: %.2f lv.\n", priceDoubleWithDiscount);
            System.out.printf("Suite: %.2f lv.\n", priceSuiteWithDiscount);

        } catch (
                Exception ex)

        {
            System.out.println("Sorry, invalid input!!! Start from the beginning, please! " +
                    "Read the instructions carefully!");
        }
    }
}
